Source/WebCore/loader/PolicyChecker.cpp

```diff
@@ -76,19 +77,11 @@ static bool isAllowedByContentSecurityPolicy(const URL& url, const Element* owne
     return ownerElement->document().contentSecurityPolicy()->allowChildFrameFromSource(url, redirectResponseReceived);
 }
 
-PolicyCheckIdentifier PolicyCheckIdentifier::create()
+static bool shouldExecuteJavaScriptURLSynchronously(const URL& url)
 {
-    static uint64_t identifier = 0;
-    identifier++;
-    return PolicyCheckIdentifier { Process::identifier(), identifier };
-}
-
-bool PolicyCheckIdentifier::isValidFor(PolicyCheckIdentifier expectedIdentifier)
-{
-    RELEASE_ASSERT_WITH_MESSAGE(m_policyCheck, "Received 0 as the policy check identifier");
-    RELEASE_ASSERT_WITH_MESSAGE(m_process == expectedIdentifier.m_process, "Received a policy check response for a wrong process");
-    RELEASE_ASSERT_WITH_MESSAGE(m_policyCheck <= expectedIdentifier.m_policyCheck, "Received a policy check response from the future");
-    return m_policyCheck == expectedIdentifier.m_policyCheck;
+    // FIXME: Some sites rely on the javascript:'' loading synchronously, which is why we have this special case.
```

Comment:
Is this tested by WPT?

Reply:
I am not sure about WPT test coverage, we do have layout test coverage though. This is also not new behavior in this patch, just moving it.